pkg/provisioner/crdprovisioner.go

```diff
@@ -395,92 +467,42 @@ func (c *CrdProvisioner) PublishVolume(
 	}
 	defer waiter.Close()
 
-	if volumeContext == nil {
-		volumeContext = map[string]string{}
-	}
 	azVolumeAttachmentInstance, err := lister.Get(attachmentName)
-	if err == nil {
-		// if CRI is scheduled for deletion, retry until operations are complete
-		if azVolumeAttachmentInstance.DeletionTimestamp != nil {
-			return nil, status.Errorf(codes.Aborted, "need to wait until attachment is fully deleted for volume (%s) and node (%s) before attaching", volumeName, nodeID)
-		}
-		// If there already exists a primary attachment with populated responseObject return
-		if azVolumeAttachmentInstance.Status.Detail != nil && azVolumeAttachmentInstance.Status.Detail.PublishContext != nil && azVolumeAttachmentInstance.Status.Detail.Role == v1alpha1.PrimaryRole {
-			return azVolumeAttachmentInstance.Status.Detail.PublishContext, nil
-			// if attachment is pending return error to prevent redundant attachment request
-		} else if azVolumeAttachmentInstance.Status.State == v1alpha1.Attaching {
-			return nil, status.Errorf(codes.Aborted, "attachment still in process for volume (%s) and node (%s)", volumeName, nodeID)
-		}
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "unexpected error while getting AzVolumeAttachment (%s): %v", attachmentName, err)
+	}
 
+	// If the AzVolumeAttachment is attached, return without triggering informer wait
```

Comment:
Shouldn't we wait regardless to ensure the `attach_detach` controller has processed the promotion request so that `AzVolumeAttachment` is fully correct before proceeding?

Reply:
I think that wait should be done in `WaitForAttach` instead and have nodeserver check whether `AzVolumeAttachment` is primary or not before proceeding so that `WaitForAttach` can be called appropriately.